azurerm/internal/services/authorization/role_definition_resource.go

```diff
@@ -217,7 +222,9 @@ func resourceArmRoleDefinitionDelete(d *schema.ResourceData, meta interface{}) e
 	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := parseRoleDefinitionId(d.Id())
+	scope := d.Get("scope").(string)
```

Comment:
we can't retrieve fields from the state during the read (there's no guarantees they're present) - so we'll need to pull this from somewhere else (e.g. look it up / put it in the ID?)

Reply:
this was fixed by @jackofallops in `azurerm/internal/services/authorization/parse/role_definition.go` so the scope is saved in the ID